app/src/validators/common.js

```diff
@@ -0,0 +1,40 @@
+const { Joi } = require('express-validation');
+const { Permissions } = require('../components/constants');
+
+const alphanumModel = Joi.string().alphanum().max(255);
+
+const truthyModel = Joi.boolean()
+  .truthy('true', 1, '1', 't', 'yes', 'y', 'false', 0, '0', 'f', 'no', 'n');
+
+const uuidv4 = Joi.string().guid({
+  version: 'uuidv4'
+});
+
+const uuidv4MultiModel =  Joi.alternatives().try(
+  Joi.array().items(uuidv4),
+  uuidv4
+);
+
+const stringMultiModel = Joi.alternatives().try(
+  Joi.array().items(Joi.string().max(255)),
+  Joi.string().max(255)
+);
+
+const permCodeMultiModel = Joi.alternatives().try(
+  Joi.array().items(Joi.string().max(255).valid(
+    Permissions.CREATE,
+    Permissions.READ,
+    Permissions.UPDATE,
+    Permissions.DELETE,
+    Permissions.MANAGE
+  )),
+  Joi.string().max(255).valid(
+    Permissions.CREATE,
+    Permissions.READ,
+    Permissions.UPDATE,
+    Permissions.DELETE,
+    Permissions.MANAGE
+  )
```

Comment:
You could do something clever with asserting permissions with an array spread operator like the following:
```js
Joi.array().items(Joi.string().max(255).valid(...Permissions),
Joi.string().max(255).valid(...Permissions)
```
Since we already define Permissions object, and we're effectively using all of the object, we can just let JS handle the de-structuring assignment. This has the added benefit of allowing us to need to refactor less in the event we add a new permissions constant, as it'd just immediately propagate here with spread syntax instead of needing us to explicitly name it.